### PR TITLE
Set up MongoDB and basic models with MVC structure

### PR DESCRIPTION
**Pull request by ashleywd**
This PR replaces #2 which was closed due to force push issues.

@daosgava Did you want to re-review and add any new feedback or all good to merge?

The one outstanding comment that I did not apply was about adding points which can be resolved via another card. Should have points to post, however all users will need to start with some points for it to work. I can address it now or we can do it later.

### CODE REVIEW

File: config/db.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    try {
+        await mongoose.connect(mongoURI);
+        console.log('Connected to MongoDB successfully');
+    } catch (err) {
+        console.error('MongoDB connection error:', err);
+        process.exit(1);
+    }
+};
+
+module.exports = { connectDB };
```
Comment (daosgava) [inline config/db.js:14]: Good :)

File: controllers/controller.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        console.error(err.stack);
+        res.status(500).json({ error: 'Something went wrong!' });
+    },
+
+    handle404: (req, res) => {
+        res.status(404).json({ error: 'Route not found' });
+    }
+};
+
+module.exports = controller;
```
Comment (daosgava) [inline controllers/controller.js:30]: One minor detail is that this file should be appController.js, so we have the model, view, and controller for each functionality. For example, for a post, we will have postController, postModel, and its corresponding view. The same applies to the other features.

File: routes/app.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const router = express.Router();
+const controller = require('../controllers/controller');
+
+// Basic route handlers
+router.get('/', controller.renderHome);
+router.get('/posts', controller.renderPosts);
+router.get('/login', controller.renderLogin);
+router.get('/register', controller.renderRegister);
+
```
Comment (daosgava) [inline routes/app.js:10]: Nice!

File: test/seed.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            });
+            await testPost.save();
+            console.log('Test post created successfully');
+        }
+    } catch (error) {
+        console.error('Error creating test data:', error);
+    }
+};
+
+module.exports = { createTestData };
```
Comment (daosgava) [inline test/seed.js:33]: I am not sure about this test. Is it for entering random data into the database? If it is for unit testing, we should definitely use Jest. Here is an example: https://jestjs.io/docs/mongodb. If it’s for testing, I think we should add it when working on the feature.

**Review by daosgava — COMMENTED** [review]
@ashleywd Thanks for the changes, they look good. I left a couple of comments. Let me know if you'd like to do some pair programming so I can elaborate on my points. By the way, you can replay directly on my messages :)

File: server.js [hunk 5]
```diff
@@ -1,59 +1,37 @@
...
-    resave: false,
-    saveUninitialized: false
-}));
+app.use(express.static(path.join(__dirname, 'public')));
+app.use('/src', (req, res, next) => {
+    if (req.path.endsWith('.js')) {
+        res.type('application/javascript');
+    }
+    next();
+}, express.static(path.join(__dirname, 'src')));
```
Comment (daosgava) [inline server.js:22]: By the way, we don't need this. We should not give access to the src folder, only to the public folder.

File: controllers/controller.js (hunk 2 above)
Comment (ashleywd) [reply controllers/controller.js:30]: Great, will change controller.js to appController.js

File: test/seed.js (hunk 4 above)
Comment (ashleywd) [reply test/seed.js:33]: No worries, I will just delete this file.

File: server.js (hunk 5 above)
Comment (ashleywd) [reply server.js:22]: Good feedback. Deleting these lines entirely.

**Review by daosgava — APPROVED** [review]
@ashleywd This is good to go :). I don’t think we need to worry about the models and properties at this stage. We can add them as we start working on these features, and it will likely become clearer what needs to be added or changed as we move forward.

**Pull request merged**